test/framework/src/main/java/org/elasticsearch/test/rest/client/RestClient.java

```diff
@@ -284,7 +288,42 @@ protected CloseableHttpClient createHttpClient(Settings settings) throws IOExcep
                 SSLContext sslcontext = SSLContexts.custom()
                         .loadTrustMaterial(keyStore, null)
                         .build();
-                sslsf = new SSLConnectionSocketFactory(sslcontext);
+                sslsf = new SSLConnectionSocketFactory(sslcontext, new X509HostnameVerifier() {
```

Comment:
Can we just have a single shared instance? Can this be extracted into a separate class and have tests added?